ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -221,6 +223,7 @@ public void testConcurrentPutsSuccess()
     for (int i = 0; i < 5; i++) {
       requestAndResultsList.add(new RequestAndResult(chunkSize + random.nextInt(5) * random.nextInt(chunkSize)));
     }
+    instantiateNewRouterForPuts = false;
```

Comment:
a comment couple of lines above says
>  do more puts on the same router (that is not closed).

Wouldn't this contravene that comment? What do you think about asserting that the `Router` is not closed at the point you make that comment? That way we fail fast if the assumption is incorrect.